api/swagger.yaml

```diff
@@ -487,6 +487,41 @@ definitions:
         type: "integer"
         format: "int64"
 
+  ResourceObject:
+    description: "An object describing the resources which can be advertised by a node and requested by a task"
+    type: "object"
+    properties:
+      NanoCPUs:
+        type: "integer"
+        format: "int64"
+      MemoryBytes:
+        type: "integer"
+        format: "int64"
+      GenericResources:
+        $ref: "#/definitions/GenericResources"
+
+  GenericResources:
+    description: "User defined Resources, can be either Integer resources (e.g: SSD=3) or String resources (e.g: GPU={UUID1, UUID2})"
+    type: "array"
+    items:
+      type: "object"
+      properties:
+        NamedResourceSpec:
```

Comment:
This looks very different from what is in swarmkit. https://github.com/docker/swarmkit/blob/d4a01ccd09df3e554d9efc670869fd130ed0154e/api/types.proto#L38